tests/chainer_tests/functions_tests/connection_tests/test_deconvolution_nd.py

```diff
@@ -52,91 +48,113 @@
-        self.check_double_backward_options = {
-            'dtype': numpy.float64, 'atol': 5e-3, 'rtol': 5e-2}
+        self.N = 2
+        self.in_channels = 4
+        self.out_channels = 2
+        self.ndim = len(self.dims)
+        self.ksize = (3,) * self.ndim
+        self.stride = (2,) * self.ndim
+        self.pad = (1,) * self.ndim
+        self.dilate = (self.dilate,) * self.ndim
+
+        self.W_scale = numpy.sqrt(1. / functools.reduce(mul, self.ksize,
+                                                        self.in_channels))
+        self.W_shape = (self.in_channels,
+                        self.out_channels // self.groups) + self.ksize
 
         outs = tuple(
             conv.get_deconv_outsize(d, k, s, p, d=di)
             for (d, k, s, p, di)
-            in zip(self.dims, ksize, self.stride, self.pad, self.dilate))
+            in zip(self.dims, self.ksize, self.stride, self.pad, self.dilate))
         self.outsize = outs if self.test_outsize else None
-        x_shape = (N, in_channels) + self.dims
-        self.x = numpy.random.uniform(-1, 1, x_shape).astype(self.x_dtype)
-        gy_shape = (N, out_channels) + outs
-        self.gy = numpy.random.uniform(-1, 1, gy_shape).astype(self.x_dtype)
-
-        self.ggx = numpy.random.uniform(
-            -1, 1, self.x.shape).astype(self.x.dtype)
-        self.ggW = numpy.random.uniform(
-            -1, 1, self.W.shape).astype(self.W.dtype)
-        self.ggb = numpy.random.uniform(
-            -1, 1, self.b.shape).astype(self.b.dtype)
-
-        self.test_forward_options = {}
-        self.check_backward_options = {
-            'dtype': numpy.float64, 'atol': 3e-5, 'rtol': 3e-4}
+        self.x_shape = (self.N, self.in_channels) + self.dims
+        self.gy_shape = (self.N, self.out_channels) + outs
+
+        self.check_backward_options.update({'atol': 3e-5, 'rtol': 3e-4})
+        self.check_double_backward_options.update({'atol': 5e-3, 'rtol': 5e-2})
         if (self.x_dtype == numpy.float16 or self.W_dtype == numpy.float16
                 or self.b_dtype == numpy.float16):
-            self.test_forward_options = {'atol': 5e-4, 'rtol': 5e-3}
-            self.check_backward_options = {
-                'dtype': numpy.float64, 'atol': 2 ** -4, 'rtol': 2 ** -4}
-
-    def check_forward_consistency(self, use_cudnn='always'):
-        x_cpu = chainer.Variable(self.x)
-        W_cpu = chainer.Variable(self.W)
-        b_cpu = None if self.nobias else chainer.Variable(self.b)
-        y_cpu = F.deconvolution_nd(
-            x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+            self.check_forward_options.update({'atol': 5e-4, 'rtol': 5e-3})
+            self.check_backward_options.update({
+                'atol': 2 ** -4, 'rtol': 2 ** -4})
+            self.check_double_backward_options.update({
+                'atol': 2 ** -4, 'rtol': 2 ** -4})
+
+    def before_test(self, test_name):
+        self.backend_config.autotune = self.autotune
```

Comment:
Let's parameterize `autotune` using `testing.inject_backend_tests`, not this way.
The number of combinations will be increased. If you think that's problematic, you can `raise SkipTest` selectively.